backend/kernelCI_app/urls.py

```diff
@@ -1,18 +1,29 @@
 from django.urls import path
+from django.views.decorators.cache import cache_page
+from django.conf import settings
 from kernelCI_app import views
 
+timeout = settings.CACHE_TIMEOUT
 
 urlpatterns = [
-    path("tree/", views.TreeView.as_view(), name="tree"),
-    path("tree/<str:commit_hash>/commits", views.TreeCommitsHistory.as_view(), name="treeCommits"),
-    path("tree/tests/", views.groupedTests.as_view(), name="treeGroupedTests"),
-    path("tree/<str:commit_hash>", views.TreeDetails.as_view(), name="treeDetails"),
+    path("tree/", cache_page(timeout)
+         (views.TreeView.as_view()), name="tree"),
+    path("tree/<str:commit_hash>/commits", cache_page(timeout)
+         (views.TreeCommitsHistory.as_view()), name="treeCommits"),
+    path("tree/tests/", cache_page(timeout)
+         (views.groupedTests.as_view()), name="treeGroupedTests"),
+    path("tree/<str:commit_hash>", cache_page(timeout)
+         (views.TreeDetails.as_view()), name="treeDetails"),
     path(
-        "tree/<str:commit_hash>/tests/", views.TreeTestsView.as_view(), name="treeTests"
+        "tree/<str:commit_hash>/tests/",
+        cache_page(timeout)(views.TreeTestsView.as_view()), name="treeTests"
     ),
-    path("build/<str:build_id>", views.BuildDetails.as_view(), name="buildDetails"),
```

Comment:
I'd rather write this way:
```
path("tree/", cache_page(timeout)(views.TreeView.as_view()), 
    name="tree"),
```
In order to improve readability.
The rest is perfect!